fsspec/registry.py

```diff
@@ -59,7 +59,7 @@ def get_filesystem_class(protocol):
             raise RuntimeError(str(err))
         registry[protocol] = getattr(mod, name)
     cls = registry[protocol]
-    if cls.protocol == 'abstract' or cls.protocol is None:
+    if getattr(cls, 'protocol', None) in ('abstract', None):
```

Comment:
The class will always have a protocol attribute, so long as it derives from AbstractFileSystem (which has "abstract")

Reply:
Well at least currently `S3FileSystem` is not derived from `AbstractFileSystem`, which is how I stumbled on this.

Reply:
https://github.com/dask/s3fs/pull/161

Reply:
(you should not use any of the code in fsspec with anything *not* derived or otherwise completely compatible with AbstractFileSystem)

Reply:
As things stand, versions `0.1.2` and `0.1.3` are broken, at least for s3:
```
$ python -c "import fsspec; fsspec.filesystem('s3')"
...
AttributeError: type object 'S3FileSystem' has no attribute 'protocol'
```
The patch fixes it and is backwards compatible; even when dask/s3fs#161 lands it doesn't hurt to leave the fix in.

Reply:
The S3FileSystem that you get without the WIP PR, though, won't actually function as you expect given the fsspec spec. I guess your change can't hurt anything, it's true, but what are you wanting to do? I haven't checked, so I'm not sure that basic things like `open_files` would work with the pre-PR version.

Reply:
(but yes, you have persuaded me that this line can change as you suggest)

Reply:
(btw I added `isfile` and `isdir` to `s3fs` in https://github.com/gsakkis/s3fs/commit/3b119ef041d655bbec15a5aca807e4163e12a9d5; I think that's needed even after https://github.com/dask/s3fs/pull/161 because the default `AbstractFileSystem` implementation cannot be used there)

Reply:
Why cannot it be used? 
You seem to have some code interacting with these implementations, is that public?

Reply:
Because ` self.info(path)` raises an exception in S3 if path is a directory (and rightly so), so at least `isdir` needs to be implemented otherwise.

Reply:
My code is a simple [Path](https://docs.python.org/3/library/pathlib.html#concrete-paths)-like implementation; it's not generic and mature enough to release it as a general library or a PR to fsspec (not yet at least) but I could post it as a gist if interested.